pkg/cli/admin/node/logs.go

```diff
@@ -258,37 +287,44 @@ func (o LogsOptions) RunLogs() error {
 		req := client.Get().RequestURI(path).
 			SetHeader("Accept", "text/plain, */*").
 			SetHeader("Accept-Encoding", "gzip")
-		if o.Path == "journal" {
-			if len(o.UntilTime) > 0 {
-				req.Param("until", o.UntilTime)
+		if o.Path == "journal" || len(o.Query) > 0 {
+			if len(o.Until) > 0 {
+				req.Param("until", o.Until)
 			}
-			if len(o.SinceTime) > 0 {
-				req.Param("since", o.SinceTime)
+			if len(o.Since) > 0 {
+				req.Param("since", o.Since)
 			}
 			if len(o.Output) > 0 {
 				req.Param("output", o.Output)
 			}
 			if o.BootChanaged {
 				req.Param("boot", fmt.Sprintf("%d", o.Boot))
 			}
-			if len(o.Units) > 0 {
+			if len(o.Query) > 0 {
+				for _, unit := range o.Query {
+					req.Param("query", unit)
+				}
+			} else if len(o.Units) > 0 {
 				for _, unit := range o.Units {
-					req.Param("unit", unit)
+					req.Param("query", unit)
 				}
 			}
 			if len(o.Grep) > 0 {
-				req.Param("grep", o.Grep)
+				req.Param("pattern", o.Grep)
 				req.Param("case-sensitive", fmt.Sprintf("%t", o.GrepCaseSensitive))
```

Comment:
Do we use `case-sensitive` flag in new API?. For example, when we deprecate `grep` and starts using `pattern` in next release, can we also deprecate `case-sensitive`?

Reply:
Thanks